plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchTaskHandler.groovy

```diff
@@ -712,7 +713,7 @@ class AwsBatchTaskHandler extends TaskHandler implements BatchHandler<String,Job
         final opts = getAwsOptions()
         final labels = task.config.getResourceLabels()
         final result = new SubmitJobRequest()
-        result.setJobName(normalizeJobName(task.name))
+        result.setJobName(TaskUtils.includeTowerPrefix(normalizeJobName(task.name), this.environment))
```

Comment:
I've move `includeTowerPrefix` in `normalizeJobName`, so it will be easier to test

Reply:
I'll give a try to this